crates/cargo-remove/src/cargo/ops/cargo_remove/mod.rs

```diff
@@ -5,88 +5,60 @@ mod manifest;
 mod metadata;
 mod util;
 
-pub use dependency::Dependency;
-pub use dependency::PathSource;
-pub use dependency::RegistrySource;
-pub use dependency::Source;
-pub use manifest::{find, get_dep_version, set_dep_version, LocalManifest, Manifest};
+use cargo::core::Package;
+use cargo::CargoResult;
+
+pub use dependency::{Dependency, PathSource, RegistrySource, Source};
+pub use manifest::{
+    find, get_dep_version, set_dep_version, DepKind, DepTable, LocalManifest, Manifest,
+};
 pub use metadata::{manifest_from_pkgid, resolve_manifests, workspace_members};
 pub use util::{
     colorize_stderr, shell_note, shell_print, shell_status, shell_warn, shell_write_stderr, Color,
     ColorChoice,
 };
 
-use cargo::CargoResult;
-
-use std::borrow::Cow;
-use std::path::PathBuf;
-
 /// Remove a dependency from a Cargo.toml manifest file.
 #[derive(Debug)]
-pub struct RmOptions {
-    /// Dependencies to be removed
-    pub crates: Vec<String>,
-    /// Remove as development dependency
-    pub dev: bool,
-    /// Remove as build dependency
-    pub build: bool,
-    /// Remove as dependency from the given target platform
-    pub target: Option<String>,
-    /// Path to the manifest to remove a dependency from
-    pub manifest_path: Option<PathBuf>,
-    /// Package to remove from
-    pub pkg_id: Option<String>,
-    /// Don't actually write the manifest
+pub struct RmOptions<'a> {
+    /// Package to remove dependencies from
+    pub spec: &'a Package,
+    /// Dependencies to remove
+    pub dependencies: Vec<&'a String>,
```

Comment:
nit: probably don't need a borrow for this.  The inconsistency feels a bit weird

Reply:
Didn't seem worth handling in this PR.  Feel free to make another